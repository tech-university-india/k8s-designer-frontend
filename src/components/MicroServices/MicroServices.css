.microservices {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-;
  border: 1px solid black;
  height: 100vh;
  width: 24vw;
  min-width: 200px;
}

.microservices-header{
  height: 5vh;
  font-size: 20px;
  border-bottom : 2px solid rgb(82, 81, 81);
  text-align: center;
  padding-top: 2vh;
  width: 100%;
}
.heading {
  font-size: medium;
  display: flex;
  flex-direction: column;
}

hr {
  width: 100%;
  border: solid black;
}

.services {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
}

.service {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100px;
  width: 100px;
  border: 1px solid black;
  margin: 20px 0;
  border-radius: 20px;
  font-weight: bold;
  box-shadow: 2px 2px 2px gray;
}