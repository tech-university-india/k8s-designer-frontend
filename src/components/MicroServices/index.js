import React from 'react';
import './MicroServices.css';

const MicroServices = () => {
  const onDragStart = (event, nodeType) => {
    event.dataTransfer.setData('application/reactflow', nodeType);
    event.dataTransfer.effectAllowed = 'move';
  };
  return (
    <div className='microservices'>
      <div>
        <p className='heading'>MicroServices</p>
        <hr />
      </div>
      <div className='services'>
        <div draggable onDragStart={(event) => onDragStart(event, 'frontend')} className='service'>
          <p>FRONTEND</p>
        </div>
        <div draggable onDragStart={(event) => onDragStart(event, 'backend')} className='service'>
          <p>BACKEND</p>
        </div>
        <div draggable onDragStart={(event) => onDragStart(event, 'database')} className='service'>
          <p>DATABASE</p>
        </div>
      </div>
    </div>
  );
};

export default MicroServices;