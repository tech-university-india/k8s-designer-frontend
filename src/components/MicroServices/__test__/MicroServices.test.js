import React from 'react';
import { render } from '@testing-library/react';
import MicroServices from '../index';

describe('MicroServices', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<MicroServices />);
    expect(asFragment()).toMatchSnapshot();
  });
});