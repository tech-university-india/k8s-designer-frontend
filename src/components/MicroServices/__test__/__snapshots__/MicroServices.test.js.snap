// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MicroServices should render correctly 1`] = `
<DocumentFragment>
  <div
    class="microservices"
  >
    <div>
      <p
        class="heading"
      >
        MicroServices
      </p>
      <hr />
    </div>
    <div
      class="services"
    >
      <div
        class="service"
        draggable="true"
      >
        <p>
          FRONTEND
        </p>
      </div>
      <div
        class="service"
        draggable="true"
      >
        <p>
          BACKEND
        </p>
      </div>
      <div
        class="service"
        draggable="true"
      >
        <p>
          DATABASE
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
